electricitymap/contrib/capacity_parsers/OPENNEM.py

```diff
@@ -0,0 +1,153 @@
+    "coal_brown": "coal",
+    "distillate": "oil",
+    "bioenergy_biogas": "biomass",
+    "gas_ccgt": "gas",
+    "gas_wcmg": "gas",
+    "gas_recip": "gas",
+    "bioenergy_biomass": "biomass",
+}
+
+CAPACITY_URL = "https://api.opennem.org.au/facility/"
+
+
+def get_opennem_capacity_data(session: Session) -> dict:
+    r: Response = session.get(CAPACITY_URL)
+    data = r.json()
+    capacity_df = pd.json_normalize(data)
+
+    capacity_df = capacity_df.loc[capacity_df["dispatch_type"] == "GENERATOR"]
+    capacity_df = capacity_df.loc[capacity_df["status.code"] == "operating"]
+    capacity_df = capacity_df[
+        ["network_region", "capacity_registered", "fueltech.code", "created_at"]
+    ]
+    capacity_df = capacity_df.rename(
+        columns={
+            "network_region": "zone_key",
+            "capacity_registered": "value",
+            "fueltech.code": "mode",
+            "created_at": "datetime",
+        }
+    )
+
+    capacity_df["datetime"] = capacity_df["datetime"].apply(
+        lambda x: pd.to_datetime(x).replace(
+            month=1, day=1, hour=0, minute=0, second=0, microsecond=0, tzinfo=None
+        )
+    )
+    return capacity_df
+
+
+def filter_capacity_data_by_datetime(
+    data: pd.DataFrame, target_datetime: datetime
+) -> pd.DataFrame:
+    """Filter capacity data by datetime. For a given target_datetime, the capacity should only include plants created before then."""
+    df = data.copy()
+    max_datetime = df["datetime"].max()
+    min_datetime = df["datetime"].min()
+
+    if target_datetime >= max_datetime:
+        df = df.copy()
+    elif target_datetime <= min_datetime:
+        df = df.loc[
+            df["datetime"] == min_datetime
+        ].copy()  # we backfill the capacity data using the first data point
+    else:
+        df = df.loc[df["datetime"] <= target_datetime]
+    return df
+
+
+def fetch_production_capacity_for_all_zones(
+    target_datetime: datetime, session: Session | None = None
+):
+    if session is None:
+        session = Session()
+    capacity_df = get_opennem_capacity_data(session)
+    capacity_df = filter_capacity_data_by_datetime(capacity_df, target_datetime)
+
+    capacity_df["zone_key"] = capacity_df["zone_key"].map(REGION_MAPPING)
+    capacity_df["mode"] = capacity_df["mode"].map(FUEL_MAPPING)
+
+    capacity_df = (
+        capacity_df.groupby(["zone_key", "mode"])[["value"]].sum().reset_index()
+    )
+
+    capacity = {}
+    for zone in capacity_df["zone_key"].unique():
+        zone_capacity_df = capacity_df.loc[capacity_df["zone_key"] == zone]
+        zone_capacity = {}
+        for idx, data in zone_capacity_df.iterrows():
+            zone_capacity[data["mode"]] = {
+                "datetime": target_datetime.strftime("%Y-%m-%d"),
+                "value": round(data["value"], 2),
+                "source": SOURCE,
+            }
+        capacity[zone] = zone_capacity
+    return capacity
+
+
+def fetch_production_capacity(
```

Comment:
Same as for ONS, shouldn't we have session as an optional argument here as well?